CondCore/L1TPlugins/plugins/L1TMuonGlobalParams_PayloadInspector.cc

```diff
@@ -0,0 +1,108 @@
+/*!
+  \file L1TMuonGlobalParams_PayloadInspector
+  \Payload Inspector Plugin for L1TMuonGlobalParams payloads
+  \author Y. Chao
+  \version $Revision: 1.0 $
+  \date $Date: 2024/05/15 12:00:00 $
+*/
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+
+// the data format of the condition to be inspected
+#include "CondFormats/L1TObjects/interface/L1TMuonGlobalParams.h"
+
+#include "L1Trigger/L1TMuon/interface/L1TMuonGlobalParamsHelper.h"
+#include "L1Trigger/L1TMuon/interface/L1TMuonGlobalParams_PUBLIC.h"
+
+#include <memory>
+#include <sstream>
+#include <iostream>
+#include <bitset>
+
+// include ROOT
+#include "TH1F.h"
+#include "TLegend.h"
+#include "TCanvas.h"
+#include "TLatex.h"
+
+namespace {
+
+  using namespace cond::payloadInspector;
+
+  class L1TMuonGlobalParamsInputBits : public PlotImage<L1TMuonGlobalParams, SINGLE_IOV> {
+  public:
+    L1TMuonGlobalParamsInputBits() : PlotImage<L1TMuonGlobalParams, SINGLE_IOV>("L1TMuonGlobalParams plot") {}
+
+    bool fill() override {
+      auto tag = PlotBase::getTag<0>();
+      auto iov = tag.iovs.front();
+
+      std::string IOVsince = std::to_string(std::get<0>(iov));
+      std::shared_ptr<L1TMuonGlobalParams> payload = fetchPayload(std::get<1>(iov));
+      if (payload.get()) {
+        /// Create a canvas
+        edm::LogInfo("L1TMG") << "absIsoCheckMemLUTPath: " << payload->absIsoCheckMemLUTPath();
+        TCanvas canvas("L1TMuonGlobal", "L1TMuonGlobal", 800, 600);
+
+        L1TMuonGlobalParams l1tmg = (L1TMuonGlobalParams)*payload;
+        L1TMuonGlobalParamsHelper l1tmgph(l1tmg);
+
+        canvas.cd();
+        canvas.Update();
+
+        TLatex tl;
+        // Draw the columns titles
+        tl.SetTextAlign(12);
+        tl.SetTextSize(0.03);
+
+        TH1F* input1 = new TH1F("InputsToDisable", "", 75, 0, 75);
+        TH1F* input2 = new TH1F("MaskedInputs", "", 75, 0, 75);
+
+        TLegend* leg = new TLegend(0.60, 0.65, 0.85, 0.85);
+
+        leg->AddEntry(input1, "InputsToDisable", "l");
+        leg->AddEntry(input2, "MaskedInputs", "l");
+        leg->SetLineColor(0);
+        leg->SetFillColor(0);
+
+        input1->SetStats(0);
+        input1->SetMaximum(5);
+        input1->SetXTitle("InputBits");
+        input1->SetLabelOffset(0.9, "Y");
+        input1->SetLineWidth(3);
+        input1->SetLineColor(9);
+        input2->SetLineWidth(3);
+        input1->SetLineColor(8);
+
+        for (size_t idx = 1; idx <= 75; idx++) {
```

Comment:
what's this 75 magic number? can it be taken from elsewhere or be a named constants?

Reply:
It should be 72 for the size of input-bits in ​[L1TMuonGlobalParamsHelper.h](https://cmssdt.cern.ch/lxr/source/L1Trigger/L1TMuon/interface/L1TMuonGlobalParamsHelper.h). I'm fixing this in the later update.